### ISSUE TITLE
[search] Exact match items are ranked lower than less relevant results

### ISSUE THREAD

**Issue by cnotin** [organicmaps/organicmaps#7370]
**Describe the issue**
In this example I'm searching for the "forêt de chantilly" locations (in France). I know that this exists (there are actually several items with that name). But it isn't returned in the first results:
![IMG_9769](https://github.com/organicmaps/organicmaps/assets/550823/d0ce8cd1-a4e7-4698-aab5-fc9f0715b489)

I need to scroll down to finally see them:
![IMG_9770](https://github.com/organicmaps/organicmaps/assets/550823/003d86d2-fd06-460f-b919-dc2f1e72d8b6)

**Steps to reproduce**
1. Download maps for France -> Île de France
2. Search for "forêt de chantilly"
3. See results

**Expected behaviour**
The 3x "Forêt de Chantilly" results are shown first since they are matching exactly with my query

**Obtained behaviour**
The first results are about "Chantilly" indeed, but the "Forêt de" part is ignored and I have to scroll down to finally see the exact matches that exist

**Screenshots**
See above

**System information:**
- iOS 17.3.1
 - Organic Maps version: 2024.02.06-13
 - Device Model: iPhone

**Additional context**
Are they ranked lowed perhaps their type is "forest"?

**Comment by biodranik** [comment]
@vng I assume that "de" word is ignored by the search, correct?

If yes, then this is not the right approach.

**Comment by cnotin** [comment]
For additional context, here the "de" in French means the same as "of" in English. So I can understand why in some cases it can be ignored, but I’d say except if there’s an exact match

**Comment by vng** [comment]
Yes, ranking is bad here indeed. The main point is that there are some "Chantilly" cities and "XXX Foret" villages nearby.
The app prefers ranking a bunch of City/Village + POI matching instead of full forest "forêt de chantilly" match, because _forest_ is not a _popular_ type ...

**Comment by cnotin** [comment]
Oh I didn't know that the type popularity was taken into account 😯

**Comment by vng** [comment]
The ranking is bad, anyway. We should penalize more cases: POI _near_ Village and POI _near_ Street.

**Comment by biodranik** [comment]
@vng the full match should always be in the top.

**Comment by oleg-rswll** [comment]
Related to #10095

**Comment by cnotin** [comment]
While working on a fix I tried to reproduce and the current version ⁦2026.07.23-6⁩ behaves better:

<img width="225" height="454" alt="Image" src="https://github.com/user-attachments/assets/3f3c0b9a-7d1e-47d7-b0e4-af550cc3555b" />

The first result is the expected one. So it was perhaps fixed/improved in the meantime?

**Comment by biodranik** [comment]
Ranking depends on map viewport and the current position too. Could it change results?

**Comment by cnotin** [comment]
I thought about that, and honestly I don’t remember very well from where I made the original search, but the distance is exactly the same from the original capture so I’m pretty sure I was in the same location. Doesn’t answer the viewpoint question though. And my download added cache is the same: entire metropolitan France.
